### PR TITLE
add option to show/hide controls in material stepper

### PR DESCRIPTION
**Pull request by ricardobrg**
Material Stepper doesn't have an option to customize navigation controls "Continue"and "Cancel". This first patch is intended to give the developer an option to remove the default controls in order to create its own controls. #20237

### CODE REVIEW

**Comment by ricardobrg** [comment]
I signed it!

**Comment by Hixie** [comment]
An alternative approach would be to provide a builder to create the controls.

Should we do anything to support not showing the horizontal stepper controls?

This will need a test.

**Comment by Hixie** [comment]
Thanks for your contribution by the way!

**Comment by ricardobrg** [comment]
@Hixie thank you for all the feedback. It was just a quick fix I've made for a particular project and it need improvements. I'll work on it to implement the showControls option.

> An alternative approach would be to provide a builder to create the controls.
I believe that the builder would be the best approach. It should include the option to build horizontal and vertical controls too, implementing #11133. I'd like to work on it as soon as I fix the code and add the tests in this PR.

File: packages/flutter/lib/src/material/stepper.dart [hunk 1]
```diff
@@ -160,6 +161,9 @@ class Stepper extends StatefulWidget {
   /// The index into [steps] of the current step whose content is displayed.
   final int currentStep;
 
+  /// Wheter to show navigation controls
```
[the remark this answers is not in the document]
Comment (ricardobrg) [reply packages/flutter/lib/src/material/stepper.dart]: fixed in 42e8c05

File: packages/flutter/lib/src/material/stepper.dart [hunk 2]
```diff
@@ -325,6 +329,10 @@ class _StepperState extends State<Stepper> with TickerProviderStateMixin {
   }
 
   Widget _buildVerticalControls() {
+    if (!widget.showControls){
+	return new Container();
```
[the remark this answers is not in the document]
Comment (ricardobrg) [reply packages/flutter/lib/src/material/stepper.dart]: Agree... moved the check in 42e8c05.

**Comment by Hixie** [comment]
(See also https://github.com/flutter/flutter/pull/11165 which was an earlier attempt in a similar area.)

@ricardobrg Great! Let me know when you're ready for me to review again or if you have any questions!

**Comment by Hixie** [comment]
@ricardobrg Will you be able to continue working on this? It's totally fine if not; we can take it over (though it might take longer). I'm just asking because I'm trying to clean out our review queue.

**Comment by ricardobrg** [comment]
@Hixie sorry for the late answer... looks like @jslavitz implemented the builder through #23310. Will test it and work from there to implement horizontal builder referenced in #11165.

**Pull request closed without merging**